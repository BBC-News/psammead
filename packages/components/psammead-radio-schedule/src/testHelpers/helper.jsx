/* eslint-disable react/prop-types */
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { arabic, latin } from '@bbc/gel-foundations/scripts';
import { boolean } from '@storybook/addon-knobs';
import React from 'react';
import ProgramCard from '../ProgramCard';
import RadioSchedule from '../index';

export const sentenceCase = text =>
  text
    .toLowerCase()
    .charAt(0)
    .toUpperCase() + text.substring(1);

// Will remove and clean up in future PRs
export const stateTypes = ['live', 'onDemand', 'onDemand', 'next'];
export const uniqueStates = ['live', 'onDemand', 'next'];

const getSchedule = (service, withLongSummary) => {
  const { text, articlePath, longText } = TEXT_VARIANTS[service];
  const programDurationLabel =
    service === 'arabic' ? 'المدة الزمنية' : 'Duration';

  return stateTypes.map((state, index) => ({
    id: index,
    state,
    startTime: 1566914061212,
    link: articlePath,
    brandTitle: text,
    episodeTitle: '29/01/1990',
    summary:
      withLongSummary && state === 'live'
        ? `${longText} ${longText}`
        : longText,
    duration: 'PT1H',
    durationLabel: programDurationLabel,
  }));
};

export const renderProgramCard = ({
  state,
  service = 'news',
  episodeTitle = '29/01/1990',
  duration = 'PT30M',
  durationLabel = 'Duration',
  startTime = 1566914061212,
  displaySummary = boolean('show summary', true),
}) => {
  const { text, articlePath, longText, dir, locale, timezone } = TEXT_VARIANTS[
    service
  ];
  const programDurationLabel = dir === 'rtl' ? 'المدة الزمنية' : durationLabel;
  const nextLabel = dir === 'rtl' ? 'مباشر' : 'NEXT';
  const liveLabel = dir === 'rtl' ? 'مباشر' : 'LIVE';

  return (
    <ProgramCard
      service={service}
      script={dir === 'rtl' ? arabic : latin}
      dir={dir}
      brandTitle={text}
      summary={displaySummary ? longText : null}
      episodeTitle={episodeTitle}
      duration={duration}
      startTime={startTime}
      state={state}
      link={articlePath}
      durationLabel={programDurationLabel}
      nextLabel={nextLabel}
      liveLabel={liveLabel}
      timezone={timezone}
      locale={locale}
    />
  );
};

export const renderRadioSchedule = ({
  service = 'news',
  locale = 'en-gb',
  timezone = 'Europe/London',
  script = latin,
  dir = 'ltr',
  withLongSummary = false,
  selectedService = 'news',
}) => {
  const nextLabel = dir === 'rtl' ? 'مباشر' : 'NEXT';
  const liveLabel = dir === 'rtl' ? 'مباشر' : 'LIVE';

  return (
    <RadioSchedule
      schedules={getSchedule(selectedService, withLongSummary)}
      locale={locale}
      timezone={timezone}
      script={script}
      service={service}
      nextLabel={nextLabel}
      liveLabel={liveLabel}
      dir={dir}
    />
  );
};
