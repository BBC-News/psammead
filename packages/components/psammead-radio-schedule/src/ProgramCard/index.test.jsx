import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { renderProgramCard, sentenceCase } from '../testHelpers/helper';

const stateTypes = ['live', 'next', 'onDemand'];

describe('ProgramCard', () => {
  stateTypes.forEach(state => {
    shouldMatchSnapshot(
      `should render correctly for ${state}`,
      renderProgramCard({ state, stateLabel: sentenceCase(state) }),
    );
  });

  shouldMatchSnapshot(
    `should render correctly in RTL`,
    renderProgramCard({
      state: stateTypes[0],
      stateLabel: 'مباشر',
      duration: '30:00',
      durationLabel: 'المدة الزمنية',
      service: 'arabic',
    }),
  );
});
