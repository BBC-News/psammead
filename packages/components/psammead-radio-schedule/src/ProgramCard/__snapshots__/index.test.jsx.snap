// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramCard should render correctly for live 1`] = `
.c11 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #B80000;
  display: inline-block;
  margin-right: 0.5rem;
}

.c7 {
  color: #3F3F42;
  padding: 0.5rem 0;
  display: inline-block;
  width: 100%;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c3 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:visited {
  color: #6E6E73;
}

.c3:hover .c6 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:focus .c6 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 0.0625rem solid transparent;
  height: 100%;
}

.c1 {
  padding: 0 0.5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  margin: 0;
}

.c8 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c9 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #B80000;
  outline: 0.0625rem solid transparent;
  color: #FFFFFF;
}

.c10 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 0.75rem;
  margin: 0;
}

.c12 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .c9 {
    background-color: transparent;
    outline: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <a
        class="c3"
        href="/news/articles/cn7k01xp8kxo"
      >
        <span
          role="text"
        >
          <span
            aria-hidden="true"
            class="c4"
            dir="ltr"
          >
            LIVE 
          </span>
          <span
            class="c5"
            lang="en-GB"
          >
            Live, 
          </span>
        </span>
        Could a computer ever create better art than a human?
        <span
          class="c5"
        >
          , 
          14:54
          , 
        </span>
        <span
          class="c6 c7"
        >
          29/01/1990
        </span>
      </a>
    </h3>
    <p
      class="c8"
    >
      The critic, author, poet and TV host was known for his witty commentary on international television.
    </p>
  </div>
  <div
    class="c9"
  >
    <span
      class="c10"
    >
      <svg
        aria-hidden="true"
        class="c11"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <time
      class="c12"
      datetime="PT30M"
      dir="ltr"
    >
      <span
        class="c5"
      >
         Duration 30,00 
      </span>
      <span
        aria-hidden="true"
        class=""
      >
        30:00
      </span>
    </time>
  </div>
</div>
`;

exports[`ProgramCard should render correctly for next 1`] = `
.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  color: #6E6E73;
  padding: 0.5rem 0;
  display: inline-block;
  width: 100%;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c12:hover .c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c12:focus .c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 0.0625rem solid transparent;
  height: 100%;
}

.c1 {
  padding: 0 0.5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  margin: 0;
}

.c3 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #11708C;
  display: inline-block;
  margin-right: 0.5rem;
}

.c7 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c8 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #FFFFFF;
  outline: 0.0625rem solid transparent;
  color: #11708C;
}

.c9 > svg {
  color: #11708C;
  fill: currentColor;
  width: 1.0625rem;
  height: 0.75rem;
  margin: 0;
}

.c11 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .c8 {
    background-color: transparent;
    outline: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <span
        class="c3"
        dir="ltr"
      >
        NEXT 
      </span>
      Could a computer ever create better art than a human?
      <span
        class="c4"
      >
        , 
        14:54
        , 
      </span>
      <span
        class="c5 c6"
      >
        29/01/1990
      </span>
    </h3>
    <p
      class="c7"
    >
      The critic, author, poet and TV host was known for his witty commentary on international television.
    </p>
  </div>
  <div
    class="c8"
  >
    <span
      class="c9"
    >
      <svg
        aria-hidden="true"
        class="c10"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <time
      class="c11"
      datetime="PT30M"
      dir="ltr"
    >
      <span
        class="c4"
      >
         Duration 30,00 
      </span>
      <span
        aria-hidden="true"
        class=""
      >
        30:00
      </span>
    </time>
  </div>
</div>
`;

exports[`ProgramCard should render correctly for onDemand 1`] = `
.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  color: #3F3F42;
  padding: 0.5rem 0;
  display: inline-block;
  width: 100%;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c3 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:visited {
  color: #6E6E73;
}

.c3:hover .c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:focus .c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 0.0625rem solid transparent;
  height: 100%;
}

.c1 {
  padding: 0 0.5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  margin: 0;
}

.c7 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c8 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #222222;
  outline: 0.0625rem solid transparent;
  color: #FFFFFF;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 0.75rem;
  margin: 0;
}

.c11 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .c8 {
    background-color: transparent;
    outline: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <a
        class="c3"
        href="/news/articles/cn7k01xp8kxo"
      >
        Could a computer ever create better art than a human?
        <span
          class="c4"
        >
          , 
          14:54
          , 
        </span>
        <span
          class="c5 c6"
        >
          29/01/1990
        </span>
      </a>
    </h3>
    <p
      class="c7"
    >
      The critic, author, poet and TV host was known for his witty commentary on international television.
    </p>
  </div>
  <div
    class="c8"
  >
    <span
      class="c9"
    >
      <svg
        aria-hidden="true"
        class="c10"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <time
      class="c11"
      datetime="PT30M"
      dir="ltr"
    >
      <span
        class="c4"
      >
         Duration 30,00 
      </span>
      <span
        aria-hidden="true"
        class=""
      >
        30:00
      </span>
    </time>
  </div>
</div>
`;

exports[`ProgramCard should render correctly in RTL 1`] = `
.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c6 {
  color: #3F3F42;
  padding: 0.5rem 0;
  display: inline-block;
  width: 100%;
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 1rem;
  line-height: 1.625rem;
}

.c3 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:visited {
  color: #6E6E73;
}

.c3:hover .c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:focus .c5 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 0.0625rem solid transparent;
  height: 100%;
}

.c1 {
  padding: 0 0.5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 1rem;
  line-height: 1.625rem;
  color: #222222;
  margin: 0;
}

.c7 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 1rem;
  line-height: 1.25rem;
  color: #6E6E73;
  padding-bottom: 1rem;
  margin: 0;
}

.c8 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #222222;
  outline: 0.0625rem solid transparent;
  color: #FFFFFF;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 0.75rem;
  margin: 0;
}

.c11 {
  padding-right: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.625rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .c8 {
    background-color: transparent;
    outline: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <a
        class="c3"
        href="/arabic/articles/c1er5mjnznzo"
      >
        لماذا يخجل البعض من اسم قريته في مصر؟
        <span
          class="c4"
        >
          , 
          ١٤:٥٤
          , 
        </span>
        <span
          class="c5 c6"
        >
          29/01/1990
        </span>
      </a>
    </h3>
    <p
      class="c7"
    >
      هناك وقائع عدة تتسم بالسخرية والجدل والتنمر، ضد أهل القرية الذين أصابهم الغضب والسخط مما دفعهم إلى تقديم طلب لتغيير اسم قريتهم.
    </p>
  </div>
  <div
    class="c8"
  >
    <span
      class="c9"
    >
      <svg
        aria-hidden="true"
        class="c10"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <time
      class="c11"
      datetime="PT30M"
      dir="rtl"
    >
      <span
        class="c4"
      >
         المدة الزمنية 30,00 
      </span>
      <span
        aria-hidden="true"
        class=""
      >
        30:00
      </span>
    </time>
  </div>
</div>
`;

exports[`ProgramCard should render correctly without summary 1`] = `
.c10 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.8125rem;
  height: 0.75rem;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #B80000;
  display: inline-block;
  margin-right: 0.5rem;
}

.c7 {
  color: #3F3F42;
  padding: 0.5rem 0;
  display: inline-block;
  width: 100%;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
}

.c3 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c3:hover,
.c3:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:visited {
  color: #6E6E73;
}

.c3:hover .c6 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:focus .c6 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  padding-top: 0.5rem;
  background-color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 0.0625rem solid transparent;
  height: 100%;
}

.c1 {
  padding: 0 0.5rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c2 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #222222;
  margin: 0;
}

.c8 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  padding: 0.5rem;
  background-color: #B80000;
  outline: 0.0625rem solid transparent;
  color: #FFFFFF;
}

.c9 > svg {
  color: #FFFFFF;
  fill: currentColor;
  width: 1.0625rem;
  height: 0.75rem;
  margin: 0;
}

.c11 {
  padding-left: 0.5rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.75rem;
    line-height: 1rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .c8 {
    background-color: transparent;
    outline: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      <a
        class="c3"
        href="/news/articles/cn7k01xp8kxo"
      >
        <span
          role="text"
        >
          <span
            aria-hidden="true"
            class="c4"
            dir="ltr"
          >
            LIVE 
          </span>
          <span
            class="c5"
            lang="en-GB"
          >
            Live, 
          </span>
        </span>
        Could a computer ever create better art than a human?
        <span
          class="c5"
        >
          , 
          14:54
          , 
        </span>
        <span
          class="c6 c7"
        >
          29/01/1990
        </span>
      </a>
    </h3>
  </div>
  <div
    class="c8"
  >
    <span
      class="c9"
    >
      <svg
        aria-hidden="true"
        class="c10"
        focusable="false"
        height="12px"
        viewBox="0 0 13 12"
        width="13px"
      >
        <path
          d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
        />
        <path
          d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
        />
      </svg>
    </span>
    <time
      class="c11"
      datetime="PT30M"
      dir="ltr"
    >
      <span
        class="c5"
      >
         Duration 30,00 
      </span>
      <span
        aria-hidden="true"
        class=""
      >
        30:00
      </span>
    </time>
  </div>
</div>
`;
